Author: Ferenc Wagner <wferi@niif.hu>
Description:
 Ensure that the log socket is available for the child.
 .
 Just in case syslog has been restarted, bounce the log socket before the
 chroot.

diff -Naurp tftp-hpa.orig/tftpd/tftpd.c tftp-hpa/tftpd/tftpd.c
--- tftp-hpa.orig/tftpd/tftpd.c	2010-07-10 14:05:18.284928211 +0200
+++ tftp-hpa/tftpd/tftpd.c	2010-07-10 18:07:57.876929842 +0200
@@ -869,6 +869,15 @@ int main(int argc, char **argv)
     /* Ignore SIGHUP */
     set_signal(SIGHUP, SIG_IGN, 0);
 
+    /* Make sure the log socket is still connected.  This has to be
+       done before the chroot, while /dev/log is still accessible.
+       When not running standalone, there is little chance that the
+       syslog daemon gets restarted by the time we get here. */
+    if (secure && standalone) {
+        closelog();
+        openlog(__progname, LOG_PID | LOG_NDELAY, LOG_DAEMON);
+    }
+
 #ifdef HAVE_TCPWRAPPERS
     /* Verify if this was a legal request for us.  This has to be
        done before the chroot, while /etc is still accessible. */
